src/clients/c++/image_client.cc

```diff
@@ -85,6 +86,27 @@ Preprocess(
               << std::endl;
     exit(1);
   }
+#elif CV_MAJOR_VERSION == 4
+  cv::Mat sample;
+  if ((img.channels() == 3) && (img_channels == 1)) {
+    cv::cvtColor(img, sample, cv::COLOR_BGR2GRAY);
+  } else if ((img.channels() == 4) && (img_channels == 1)) {
+    cv::cvtColor(img, sample, cv::COLOR_BGRA2GRAY);
+  } else if ((img.channels() == 3) && (img_channels == 3)) {
+    cv::cvtColor(img, sample, cv::COLOR_BGR2RGB);
+  } else if ((img.channels() == 4) && (img_channels == 3)) {
+    cv::cvtColor(img, sample, cv::COLOR_BGRA2RGB);
+  } else if ((img.channels() == 1) && (img_channels == 3)) {
+    cv::cvtColor(img, sample, cv::COLOR_GRAY2RGB);
+  } else {
+    std::cerr << "unexpected number of channels in input image or model"
+              << std::endl;
+    exit(1);
+  }
+#else
+  std::cerr << "The installed openCV version is not supported yet" << std::endl;
```

Comment:
If possible try to log out the version here "The installed openCV version <version> ..."

Reply:
OpenCV may offer backwards compatibility. Hence, I am not putting a validation check as of now.